source-postgres/discovery.go

```diff
@@ -389,6 +431,54 @@ func getPrimaryKeys(ctx context.Context, conn *pgx.Conn) (map[string][]string, e
 	return keys, err
 }
 
+const queryDiscoverSecondaryIndices = `
+SELECT r.table_schema, r.table_name, r.index_name, (r.index_keys).n, r.column_name
+FROM (
+  SELECT tn.nspname AS table_schema,
+         tc.relname AS table_name,
+	     ic.relname AS index_name,
+	     information_schema._pg_expandarray(ix.indkey) AS index_keys,
+		 a.attnum AS column_number,
+		 a.attname AS column_name
+  FROM pg_index ix
```

Comment:
In the `queryDiscoverPrimaryKeys` the `pg_index` table is written like `pg_catalog.pg_index` - does that make any difference for this query?

Reply:
Oh whoops, I meant to go back and fully qualify the table names in this query, just forgot to. Most of the time it makes no difference so I was being lazy and using the short names while composing the query, but if the user created a table named like `pg_index` or whatever in a schema with higher priority in the search path it would misbehave, so fully-qualified names are safer.